File: BlackJack/BlackJack_csharp/BlackJack/BlackJack/controller/Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BlackJack.controller
{
	class Player
	{
		public bool PlayGame(view.Console a_view, model.GameFacade a_game)
		{
			a_view.PresentInstructions();
			a_view.DisplayHands(a_game.GetDealerHand(), a_game.GetDealerScore(), a_game.GetPlayerHand(), a_game.GetPlayerScore());
			if (a_game.IsGameOver())
			{
				a_view.DisplayWinner(a_game.IsPlayerWinner());
			}
			
			view.Console.Event e;

			e = a_view.GetEvent();
			if (e == view.Console.Event.Quit)
			{
				return false;
			}
			if (e == view.Console.Event.Start)
			{
				a_game.StartNewRound();
				
			}
			if (e == view.Console.Event.Hit)
			{
				a_game.Hit();
			}
			if (e == view.Console.Event.Stand)
			{
				a_game.Stand();
			}

			return true;
		}
	}
}

File: BlackJack/BlackJack_java/controller/Player.java
package controller;

public class Player {
  
  
  public boolean playGame(view.Console a_view, model.GameFacade a_game) {
    a_view.presentInstructions(!a_game.isGameOver() && a_game.getDealerScore() > 0);
    a_view.presentDealerHand(a_game.getDealerScore(), a_game.getDealerHand());
    a_view.presentPlayerHand(a_game.getPlayerScore(), a_game.getPlayerHand());
    
    if (a_game.isGameOver() && a_game.getDealerScore() > 0) {
      a_view.presentWinner(a_game.isDealerWinner());
    }
    
    a_view.collectEvents();
    
    if (a_view.wantsToStartNewGame()) {
      a_game.start();
    } else if (a_view.wantsToHit()) {
      a_game.hit();
      
    } else if (a_view.wantsToStand()) {
      a_game.stand();
    } else if (a_view.wantsToQuit()) {
      return false;
    }
    
    return true;
  }
}